src/core/analysis/mcmc/Mcmc.cpp

```diff
@@ -554,11 +571,42 @@ void Mcmc::initializeSampler( bool prior_only )
         
         DagNode *the_node = *i;
         the_node->setMcmcMode( true );
-        the_node->setPriorOnly( prior_only );
+        
+        if (suppress_char_data)
+        {
+            std::string node_value = the_node->getValueAsString();
+                
+            /* Whenever character data are printed to the screen, the string "character matrix"
+             * is always included: this is true of both discrete data (e.g., DNA, discrete
+             * morphology) and of continuous data.
+             */
+            bool chardata_found = node_value.find("character matrix") != std::string::npos;
```

Comment:
It looks like this string "character matrix" comes from `core/datatypes/phylogenetics/characterdata/AbstractCharacterData.cpp`.  Instead of print the node value, it would be cleaner to check if the value is derived from AbstractCharacterData using dynamic_cast.

Reply:
Hmm... I'm having trouble figuring out how to access the node value for casting, since `DagNode` doesn't have a `getValue()` function like `StochasticNode` does. If you're willing to make the change yourself by any chance, please feel free to commit to the branch!

Reply:
I thought of a few approaches but none of them worked.  Essentially you are looking for the type of a node, but the DagNode class does not expose the type.  Extracting the type by printing the value is going through the back door, and would only work for some types.

We could maybe try casting from DagNode to StochasticNode<AbstractCharacterData>, but I think that StochasticNode<T> does not inherit from StochasticNode<AbstractCharacterData> even if T inherits from AbstractCharacterData.

The fact that DagNode doesn't allow looking up the type of a node makes me wonder if (a) we should expand the DagNode interface or (b) we should not be doing that.